### ISSUE TITLE
Not bug, just curiosity 

### ISSUE THREAD

**Issue by ax-anoop** [karpathy/nn-zero-to-hero#8]
Doing some research into the method of finding an optimal learning rate. 

I made the models both from scratch as the videos and also in a torch friendly way, or well using torch modules, dataloaders, optmizer, etc .. 

However something weird when running the following, which 'should' be same as code from video. The lr - loss graph is showed in im1 below. 

im2 is using code very similar to videos, i.e manually updating weights. Why are the results not the same ? Is the optimizer doing somethign different in the backend ? Over all the training is about the same, both will converge roughly at the same rate. 

```py
def findlr(model, data, test_dataloader):
    lrs = torch.linspace(0.01, 1, 1000)
    lrs = 10**torch.linspace(-3, 0, 1000)
    lri = []
    lss = []

    optim = torch.optim.SGD(model.parameters(), lr=lrs[0])
    for i in range(len(lrs)):
        for g in optim.param_groups:
            g['lr'] = lrs[i]

        x, y = next(iter(data))
        l = calcLoss(model(x), y)
        model.zero_grad()
        l.backward()
        optim.step()

        lri.append(lrs[i].item())
        lss.append(l.item())

        print(lrs[i], l.item())
    plt.plot(lri, lss)
    plt.show()
    ```
    
 im1: 
    
<img width="597" alt="image" src="https://user-images.githubusercontent.com/95486801/193437520-55d14507-867c-411f-9c9e-e11db3b9e67c.png">

im2: 
<img width="597" alt="image" src="https://user-images.githubusercontent.com/95486801/193437537-481b38c6-447a-4bf7-86ee-bffb291b737a.png">

**Comment by ax-anoop** [comment]
<img width="597" alt="Screen Shot 2022-10-02 at 12 13 56 AM" src="https://user-images.githubusercontent.com/95486801/193437584-1df0fb71-e24a-46ce-860f-03be1e4b2a60.png">

<img width="597" alt="Screen Shot 2022-10-02 at 12 14 30 AM" src="https://user-images.githubusercontent.com/95486801/193437586-a7f42f06-871e-4f04-9829-2db764189040.png">

**Comment by JonathanSum** [comment]
I am not really sure. But according to the video, I say this is related to weight. Maybe the weight was too large in the im2.

**Comment by JaredLevi18** [comment]
It could also be the gain that the weights were multiplied to, that also affects the results ( the graphs).
